File: CandyRainTolstonogova/CandyRainC++/CandyRain/Heart.cpp
#include "Heart.h"
#include <iostream>
#include <cstdlib>
#include <string>
Heart::Heart(int x, int y,   std::string sprite_path, int spd, int num, bool p_touch, bool f_touch):
    FallingObject(x, y, sprite_path, spd, num, p_touch, f_touch) {
    std::cout << "Объект Heart создан" << std::endl;
}
Heart::~Heart() {
    std::cout << "Объект Heart уничтожен" << std::endl;
}
void Heart::playerCatchHeart(int playerX, int playerY) {
    if (abs(abs(getpositionX() - playerX) < 10 && abs(getpositionY() - playerY) < 10) < 10) {
        setplayer_touch (true);
        std::cout << "Heart пойман игроком!" << std::endl;
    }
}

void Heart::plusHealth(int &health) {
    health+=1;
    std::cout << "Здоровье увеличено. Текущее здоровье: " << health << std::endl;

}

File: CandyRainTolstonogova/CandyRainJava/Heart.java
public class Heart extends FallingObject {
    
    public Heart(int x, int y, String sprite, int spd, int num, boolean pTouch, boolean fTouch) {
        super(x, y, sprite, spd, num, pTouch, fTouch);
        System.out.println("Объект Heart создан");
    }
    
    public void playerCatchHeart(int playerX, int playerY) {
        if (Math.abs(getPositionX() - playerX) < GameMain.PLAYER_SIZE && 
            Math.abs(getPositionY() - playerY) < GameMain.PLAYER_SIZE) {
            setPlayerTouch(true);
            System.out.println("Heart пойман игроком!");
        }
    }
    
    public void plusHealth(GameMain game) {
        game.setHealth(game.getHealth() + 1);
        System.out.println("Здоровье увеличено. Текущее здоровье: " + game.getHealth());
    }
}